bblocks/import_tools/imf.py

```diff
@@ -0,0 +1,175 @@
+""" """
+from __future__ import annotations
+
+import pandas as pd
+import requests
+from datetime import datetime
+from bs4 import BeautifulSoup
+import os
+from typing import Optional
+from dataclasses import field
+import warnings
+
+from bblocks.config import PATHS
+from bblocks.cleaning_tools.clean import clean_numeric_series
+from bblocks.import_tools.common import ImportData
+
+
+def _parse_sdr_links(url: str, concat_url: str) -> dict:
+    """Function to parse SDR tables from html
+
+    When called it finds the SDR tables and finds a relevant
+    date and link
+
+    Args:
+        url (str): URL to parse
+        concat_url: base url onto which a href is attached
+
+    Returns:
+        A dictionary with dates as keys and URLs as values
+    """
+
+    try:
+        content = requests.get(url).content
+    except ConnectionError:
+        raise ConnectionError(f'Could not read page: {url}')
+
+    soup = BeautifulSoup(content, 'html.parser')
+    table = soup.find_all('table')[4]
+    links = table.find_all('a')
+
+    return {link.string: f'{concat_url}{link.get("href")}' for link in links}
+
+
+def _get_tsv_url(url: str) -> str:
+    """retrieves the URL for the SDR TSV file
+
+    Args:
+        url (str): URL to parse
+
+    Returns:
+        A link to the TSV file
+    """
+
+    try:
+        content = requests.get(url).content
+    except ConnectionError:
+        raise ConnectionError(f'could not read page: {url}')
+    soup = BeautifulSoup(content, "html.parser")
+    href = soup.find_all("a", string="TSV")[0].get("href")
+
+    return f"https://www.imf.org/external/np/fin/tad/{href}"
+
+
+def _read_sdr_tsv(url: str) -> pd.DataFrame:
+    """Reads a TSV file with SDR data from the web to a pandas dataframe
+
+    Args:
+        url (str): link to the TSV file
```

Comment:
in theory, since you are type hinting, it is not necessary to add types on docstrings.